### PR TITLE
Yael

### PR DESCRIPTION
**Pull request by YaelNorman1**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -19,7 +20,7 @@ class App extends Component {
               { id : 1 , name: "B", budget: 101, img: "blue"},
               { id : 2 , name: "C", budget: 102, img: "green"},
               { id : 3 , name: "D", budget: 103, img: "orange"}],
-      loginUser: {},
+      loginUser: {id : 0 , name: "A", budget: 100, img: "red"},
```
Comment (AfikGrinstein) [inline src/App.js:23]: Extract this JSON to data.json file and read it in the constructor.

File: src/App.js [hunk 2]
```diff
@@ -134,14 +135,42 @@ class App extends Component {
     return this.state.movies.find(movie => movie["id"] === parseInt(match.params.id))
   }
 
+  updateRentedAndBudget(movie){
```
Comment (AfikGrinstein) [inline src/App.js:138]: I recommend to use ESLint next time for more readable code

File: src/App.js [hunk 3]
```diff
@@ -134,14 +135,42 @@ class App extends Component {
...
+    const loginUserId= this.state.loginUser.id;
+    const user = this.state.users.find(user => user.id === loginUserId);
```
Comment (AfikGrinstein) [inline src/App.js:140]: why are you using the getSelectedUser method?

BTW I don't expect getSelectedUser to also set the state of the loginUser

File: src/App.js [hunk 4]
```diff
@@ -134,14 +135,42 @@ class App extends Component {
...
+    
+    if (this.payOrCreditMovie(user, movie))
+      movie.isRented=!movie.isRented;
+    else
+      alert("Can't buy another movie!")
+  }
+
+  payOrCreditMovie(user, movie){
```
Comment (AfikGrinstein) [inline src/App.js:148]: It's hard to understand the method by its name.
Also I will split this method into 2 separate methods.

Add return value type (most of boolean returned value method start with is...

File: src/App.js [hunk 5]
```diff
@@ -134,14 +135,42 @@ class App extends Component {
...
   }
 
+  updateRentedAndBudget(movie){
+    const loginUserId= this.state.loginUser.id;
+    const user = this.state.users.find(user => user.id === loginUserId);
+    
+    if (this.payOrCreditMovie(user, movie))
+      movie.isRented=!movie.isRented;
+    else
+      alert("Can't buy another movie!")
```
Comment (AfikGrinstein) [inline src/App.js:145]: Sorry, we only allow renting 1 movie at a time.

File: src/components/Catalog.js [hunk 6]
```diff
@@ -1,22 +1,33 @@
...
     render() {
 
         return (
         <div>
             <LoginUser loginUser={this.props.loginUser}/>
+            <SearchInput searchInput={this.state.searchInput} handleInputSearch={this.handleInputSearch}/>
+            <br/>
             <h5>Rented Movies</h5>
-            <MovieList movies= {this.props.movies.filter(movie => movie.isRented)}/>
+            <MovieList movies= {this.props.movies.filter(movie => movie.isRented).filter(movie => movie.title.toUpperCase().includes(this.state.searchInput.toUpperCase()))} changeRentedStatus={this.props.changeRentedStatus}/>
```
Comment (AfikGrinstein) [inline src/components/Catalog.js:27]: do the filter outside of the HTML and only inject the result value

File: src/components/Catalog.js [hunk 7]
```diff
@@ -1,22 +1,33 @@
...
             <hr/>
             <h5>All Movies</h5>
-            <MovieList movies= {this.props.movies.filter(movie => !movie.isRented)}/>
-            
-
+            <MovieList movies= {this.props.movies.filter(movie => !movie.isRented).filter(movie => movie.title.toUpperCase().includes(this.state.searchInput.toUpperCase()))} changeRentedStatus={this.props.changeRentedStatus}/>
```
Comment (AfikGrinstein) [inline src/components/Catalog.js:30]: do the filter outside of the HTML and only inject the result value

File: src/components/RemoveMovieBtn.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import React, { Component } from 'react'
+import '../style/BtnDesign.css'
+
+
+export default class RemoveMovieBtn extends Component {
```
Comment (AfikGrinstein) [inline src/components/RemoveMovieBtn.js:5]: good extraction.
a better one will be to have a 1 button component that gets onclick event and button icon.

File: src/components/SearchInput.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react'
+
+export default class SearchInput extends Component {
+    
+    handleInputChang= (event)=> {
```
Comment (AfikGrinstein) [inline src/components/SearchInput.js:5]: onChange

File: src/style/BtnDesign.css [hunk 10]
```diff
@@ -0,0 +1,12 @@
+.pos{
+    margin-left: 250px;
+}
+
+/* .btn-circle.btn-sm {
```
Comment (AfikGrinstein) [inline src/style/BtnDesign.css:5]: If you comment something it's better to delete it and let the version control (git) save this history.

File: src/components/RemoveMovieBtn.js (hunk 8 above)
Comment (YaelNorman1) [reply src/components/RemoveMovieBtn.js:5]: Exactly, that's what i wanted to talk to you on Wednesday. How to implement it.

File: src/components/Catalog.js (hunk 7 above)
Comment (YaelNorman1) [reply src/components/Catalog.js:30]: both? save 2 arrays for 'true' and 'false' condition?

Comment (AfikGrinstein) [reply src/components/Catalog.js:30]: Yes.

**Pull request merged**